tests/utils/random.ts

```diff
@@ -1,18 +1,60 @@
-import { pseudoRandomRange } from "../../exports/base";
-
-/**
- * Creates a function which can be called infinite times, each call instance will yield pseudo random number in [0, 1];
- * @param initialSeed The seed.
- * @returns The generator.
- */
-export function createPseudoRandomGenerator01(initialSeed: number) {
-    const g = (function *(): Generator<number, never, never> {
-        for (let seed = initialSeed; ; ++seed) {
-            yield pseudoRandomRange(seed, 0, 1);
-        }
-    })();
-
-    return () => {
-        return g.next().value;
-    };
+export class PseudoRandomGenerator {
+    constructor(initialSeed: number) {
+        this._seed = initialSeed;
+    }
+
+    get defaultOrder() { return this._defaultOrder }
+    set defaultOrder(value) { this._defaultOrder = value; }
+
+    /** Generates number in [min, max]. */
+    public range(min: number, max: number) {
+        return min + (max - min) * this.range01();
+    }
+
+    /** Generates number in [0, 1]. */
+    public range01() {
+        return mulberry32(this._seed++);
+    }
+
+    /** Generates number in [-1, 1]. */
+    public normalized() {
+        return this.range(-1, 1);
+    }
+
+    /**
+     * Generates a finite number.
+     * @order The order(max mag).
+     */
+    public finite(order=this._defaultOrder) {
+        return this.normalized() * order;
+    }
+
+    /**
+     * Generates a positive number.
+     * @order The order(max mag).
+     */
+    public positive(order=this._defaultOrder) {
+        return this.range01() * order + order * 1e-6;
+    }
+
+    /**
+     * Generates a negative number.
+     * @order The order(max mag).
+     */
+    public negative(order=this._defaultOrder) {
+        return this.range01() * -order - order * 1e-6;
+    }
+
+    private _seed = 0;
+    private _defaultOrder = 1e3;
+}
+
+export { getMagicSeed } from './random-seed';
+
+function mulberry32(a: number) {
+    // https://stackoverflow.com/a/47593316
+    let t = a += 0x6D2B79F5;
```

Comment:
Want to do as follow:
```typescript
a += x6D2B79F5;
let t = a;
```
?

Reply:
I was completely copying and pasting the code in https://stackoverflow.com/a/47593316 .